### PR TITLE
#346 fixed bug that mob thinks there is something to commit if git configuration status.branch is true

### LINKED ISSUE #346: `mob start` breaks when git config `status.branch` is `true`
When the working directory is clean and `status.branch` is `true`, `mob start` returns the error

> &gt; cannot start; clean working tree required

I believe it is caused here, erroneously expecting `git status` to be empty: https://github.com/remotemobprogramming/mob/blob/ee1b922f846c1e9cf31cb0fd501d1a0fa2c012e4/mob.go#L1250-L1253

I believe line 1251 should be calling `git status --porcelain` instead of `git status --short`. I am happy to make a PR for that but do not have the time at the moment to learn enough Go to get an environment running or write a proper test.

### PR DESCRIPTION
**Pull request by hollesse**
closes #346

### CODE REVIEW

File: mob_test.go [hunk 1]
```diff
@@ -1995,6 +1997,8 @@ func cloneRepository(path, remoteDirectory string) {
 	git("clone", "--origin", "origin", "file://"+remoteDirectory, ".")
 	git("config", "--local", "user.name", name)
 	git("config", "--local", "user.email", name+"@example.com")
+	// see bug #346 changes the output of git status --short
+	git("config", "--local", "status.branch", "true")
```
Comment (simonharrer) [inline mob_test.go:2001]: Is now every test using this config? Or should we have a single test that makes use of this config change only?

Comment (hollesse) [reply mob_test.go:2001]: I did it for every test and that was good, as I found some more points where `--short` was used and this option let the tests for it fail.

Comment (gregorriegler) [reply mob_test.go:2001]: due to all tests having `git("config", "--local", "status.branch", "true")` now, our test-coverage is not better than it was before that. I would prefer if we had just one test that contained the line `git("config", "--local", "status.branch", "true")` to test specifically that case

Comment (hollesse) [reply mob_test.go:2001]: Do you think this is enough? There where multiple Test that failed. I would at least copy all of them then.

Comment (gregorriegler) [reply mob_test.go:2001]: Did all those tests fail of the same reason? Or did they fail for different reasons?
If they failed for the same reason, I would only pick one

Comment (hollesse) [reply mob_test.go:2001]: They failed because all used git status --short command but they did use it in different methods and for different things

**Review by gregorriegler — APPROVED** [review]

**Comment by benallred** [comment]
Would it be helpful for me to attempt a rebase of this branch and resubmit this PR?

**Pull request merged**